src/Stratis.Bitcoin/Builder/FullNodeBuilder.cs

```diff
@@ -203,7 +204,22 @@ private IServiceCollection BuildServices()
 
             // configure features startup
             foreach (var featureRegistration in this.Features.FeatureRegistrations)
+            {
+                try
+                {
+                    featureRegistration.EnsureDependencies(this.Features.FeatureRegistrations);
+                }
+                catch (MissingDependencyException e)
+                {
+                    var logger = this.NodeSettings.LoggerFactory.CreateLogger(typeof(FullNodeBuilder).FullName);
```

Comment:
Ah FYI settings has a logger already so you can just do 
`this.NodeSettings.Logger`